sky/adaptors/azure.py

```diff
@@ -55,6 +62,51 @@ def get_client(name: str, subscription_id: str):
         elif name == 'resource':
             from azure.mgmt.resource import ResourceManagementClient
             return ResourceManagementClient(credential, subscription_id)
+        elif name == 'storage':
+            from azure.mgmt.storage import StorageManagementClient
+            return StorageManagementClient(credential, subscription_id)
+        elif name == 'authorization':
+            from azure.mgmt.authorization import AuthorizationManagementClient
+            return AuthorizationManagementClient(credential, subscription_id)
+        elif name == 'graph':
+            from msgraph import GraphServiceClient
+            return GraphServiceClient(credential)
+        elif name == 'container':
+            from azure.storage.blob import ContainerClient
+            container_url = (f'https://{storage_account_name}.'
+                             f'blob.core.windows.net/{container_name}')
+            container_client = ContainerClient.from_container_url(container_url)
+            try:
+                container_client.exists()
+            except exceptions().ClientAuthenticationError as e:
+                # Caught when credential is not provided to the private
+                # container url or wrong container name is provided as the
+                # public container url. We reattempt with credentials assuming
+                # user is using private container with access rights.
+                if 'ErrorCode:NoAuthenticationInformation' in e.message:
+                    container_client = ContainerClient.from_container_url(
+                        container_url, credential)
+                    try:
+                        # Suppress noisy logs from Azure SDK when attempting
+                        # to run exists() on public container with credentials.
+                        # Reference:
+                        # https://github.com/Azure/azure-sdk-for-python/issues/9422 # pylint: disable=line-too-long
+                        azure_logger = logging.getLogger('azure')
+                        original_level = azure_logger.getEffectiveLevel()
+                        azure_logger.setLevel(logging.CRITICAL)
+                        container_client.exists()
+                        azure_logger.setLevel(original_level)
```

Comment:
@cblmemo Do you think it's not necessary to set it back to original level after? I was thinking of only silencing the noise coming from `container_client.exists()` so it can show other possible errors later that was not discovered before.

Reply:
I'm happy w/the current impl as well. Feeling the function `get_client` becomes too complicated though

Reply:
I think the art of it is to find good balance between modularity and the depth of the function.